mayastor-test/test_rebuild.js

```diff
@@ -0,0 +1,376 @@
+  var client;
+
+  var ObjectType = {
+    NEXUS: 0,
+    SOURCE_CHILD: 1,
+    DESTINATION_CHILD: 2,
+  };
+
+  function checkState(childType, expectedState) {
+    client.ListNexus({}, (err, res, done) => {
+      if (err) return done(err);
+      assert.lengthOf(res.nexus_list, 1);
+
+      let nexus = res.nexus_list[0];
+      assert.equal(nexus.uuid, UUID);
+
+      if (childType == ObjectType.NEXUS) {
+        assert.equal(nexus.state, expectedState);
+      } else if (childType == ObjectType.SOURCE_CHILD) {
+        assert.equal(nexus.children[0].state, expectedState);
+      } else if (childType == ObjectType.DESTINATION_CHILD) {
+        assert.equal(nexus.children[1].state, expectedState);
+      }
+    });
+  }
+
+  function numRebuilds(expected) {
+    client.ListNexus({}, (err, res, done) => {
+      if (err) return done(err);
+      assert.lengthOf(res.nexus_list, 1);
+
+      let nexus = res.nexus_list[0];
+      assert.equal(nexus.uuid, UUID);
+
+      assert.equal(nexus.rebuilds, expected);
+    });
+  }
+
+  function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
+
+  const createNexus = args => {
+    return new Promise((resolve, reject) => {
+      client.createNexus(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const destroyNexus = args => {
+    return new Promise((resolve, reject) => {
+      client.destroyNexus(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const addChild = args => {
+    return new Promise((resolve, reject) => {
+      client.addChildNexus(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const removeChild = args => {
+    return new Promise((resolve, reject) => {
+      client.removeChildNexus(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const startRebuild = args => {
+    return new Promise((resolve, reject) => {
+      client.startRebuild(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const stopRebuild = args => {
+    return new Promise((resolve, reject) => {
+      client.stopRebuild(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const rebuildState = args => {
+    return new Promise((resolve, reject) => {
+      client.getRebuildState(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  before(done => {
+    client = createGrpcClient('MayaStor');
+    if (!client) {
+      return done(new Error('Failed to initialize grpc client'));
+    }
+
+    async.series(
+      [
+        common.ensureNbdWritable,
+        next => {
+          fs.writeFile(child1, '', next);
```

Comment:
The writeFile uses "next" as its callback. Not sure how this would work if we were to put multiple writeFile's in the same step. 

As we discussed, we can address this at a later stage if it becomes an issue.